flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontendParser.java

```diff
@@ -129,6 +132,16 @@
                             + "You need to allow this if you removed an operator from your "
                             + "program that was part of the program when the savepoint was triggered.");
 
+    public static final Option SAVEPOINT_RESTORE_MODE =
+            new Option(
+                    "r",
```

Comment:
The short opt 'r' conflicts with `RUNNING_OPTION`, which might impact the dictint option parsing. BTW, I think `Option.builder` could be better choose to create this option.

Reply:
I don't think it is an issue. I've seen the `r` option is used for `RUNNING_OPTION`, but those options are used for different commands (list & run), which should be fine. We already do have options with the same shortcut for different contexts e.g. `s` for `SAVEPOINT_PATH_OPTION` & `SCHEDULED_OPTION` and they work just fine.

Reply:
As for the `builder` I'd prefer to keep it consistent. None of the existing options use the builder.

Reply:
I think you could refer to https://issues.apache.org/jira/browse/FLINK-22701 to see whether this is fine for all cases.

Reply:
Yes, I still think it is fine to have a clash between options for different commands. 

The problem in the task you're referring is that it clashes with `customCommandLineOptions` which are defined in custom command lines such as e.g. yarn. We should probably add a check in the `mergeOptions` method to verify there are no clashes.

As for this particular option (`r`), I could not find a clash in any of the custom command lines.